### ISSUE TITLE
Seq.unapplySeq(LazyList.from(0)).drop(1) cause infinite loop only in scalajs

### ISSUE THREAD

**Issue by seraphr** [scala-js/scala-js#3808]
Using Scala.js 0.6.29 / scala 2.13.1 / sbt 1.3.2

following code cause infinite loop and OOM error in scalajs

```scala
object Main {
  def main(args: Array[String]): Unit = {
    println("run!")
    Seq.unapplySeq(LazyList.from(0)).drop(1)
    // Stream is same
    //    Seq.unapplySeq(Stream.from(0)).drop(1)
  }
}
```

reproduction project
https://github.com/seraphr/scalajs-infinite-loop

travis log
https://travis-ci.com/seraphr/scalajs-infinite-loop/builds/131359829

**Comment by sjrd** [comment]
I think that was fixed in 2.13.1 in the upstream standard library. Scala.js is still built with the 2.13.0 version of the stdlib. This bug should fix itself once we upgrade to 2.13.1 as our reference 2.13 version.

**Comment by sjrd** [comment]
Fixed in ae27cb415c54625a99e5ebe5e7cf139bef2ca24c.
